game-app/game-core/src/main/java/games/strategy/triplea/ai/pro/ProPurchaseAi.java

```diff
@@ -427,7 +428,7 @@ private boolean shouldSaveUpForAFleet(
       return false;
     }
     // Don't save up more if we already have enough PUs to buy the biggest fleet we can.
-    IntegerMap<Resource> maxShipCost = IntegerMap.of();
+    IntegerMap<Resource> maxShipCost = IntegerMap.of(new LinkedHashMap<>());
```

Comment:
Initialize IntegerMap with a mutable implementation here to avoid immutable default implementation

Reply:
Please use the constructor instead.
```suggestion
    IntegerMap<Resource> maxShipCost = new IntegerMap<>();
```